framework/bug-mining/promote-to-db.pl

```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env perl
+#
+#-------------------------------------------------------------------------------
+# Copyright (c) 2014-2018 René Just, Darioush Jalali, and Defects4J contributors.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#-------------------------------------------------------------------------------
+
+=pod
+
+=head1 NAME
+
+promote-to-db.pl -- Promote reproducible and minimized bugs with all meta data
+to the main Defects4J database.
+
+=head1 SYNOPSIS
+
+promote-to-db.pl -p project_id -w work_dir -r repository_dir [-b bug_id] [-o output_dir] [-d output_db_dir]
+
+=head1 OPTIONS
+
+=over 4
+
+=item B<-p C<project_id>>
+
+The id of the project for which the revision pairs are to be promoted.
+
+=item B<-b C<bug_id>>
+
+Only analyze this bug id. The bug_id has to follow the format B<(\d+)(:(\d+))?>.
+Per default all bug ids, listed in the commit-db, are considered.
+
+=item B<-w C<work_dir>>
+
+The working directory used for the bug-mining process.
+
+=item B<-r C<repository_dir>>
+
+The path to the repository of this project.
+
+=item B<-o C<output_dir>>
+
+The output directory as the Defects4J directory. Defaults to F<$PROJECTS_DIR>.
+
+=item B<-d C<output_db_dir>>
```

Comment:
Do  we need this parameter? It seems like the database files are all temporary artifacts of the bug-mining process, and only the mined bugs and their meta data will be promoted to D4J proper.